### DISCUSSION TITLE
Do i need to install a skupper router in every namespace?

### DISCUSSION THREAD

**Discussion by ewassef** [skupperproject/skupper#845] [General]
It seems that if the skupper router isn't installed in the same namespace as the service/deployment, it will not expose it. Is that right? Do we need to install dozens of routers and tokens to expose multiple services?

What am I missing? the documentation is considerably lacking

**Comment by jiridanek** [comment]
No, you should be able to install skupper clusterwide if you have the `cluster-admin` access,

> Log into your cluster. If you are deploying Skupper to be available for all namespaces, make sure you have cluster-admin privileges.

[...]

> To install Skupper for all namespaces:
>
> 1. Create a namespace named skupper-site-controller.
> 2. Deploy the site controller using the following YAML:
>
> `kubectl apply -f https://raw.githubusercontent.com/skupperproject/skupper/1.0/cmd/site-controller/deploy-watch-all-ns.yaml`

> -- https://skupper.netlify.app/skupper/latest/declarative/index.html#installing-using-yaml

@pwright am I right? And does the docs say how to do it with the CLI?

**Comment by grs** [comment]
Skupper will only expose deployments in the namespace in which it is running, yes. You can have it expose a *service* in another namespace as a target (e.g. `skupper expose service other-namespace/foo`).
